subsys/bluetooth/mesh/light_ctl_srv.c

```diff
@@ -432,28 +461,20 @@ static int bt_mesh_light_ctl_srv_start(struct bt_mesh_model *model)
 
 	case BT_MESH_ON_POWER_UP_ON:
 		bt_mesh_light_temp_srv_set(&srv->temp_srv, NULL, &temp, NULL);
-		lightness_srv_change_lvl(&srv->lightness_srv, NULL, &light,
-					 NULL);
 		break;
 
 	case BT_MESH_ON_POWER_UP_RESTORE:
 		temp.params = srv->temp_srv.last;
 		bt_mesh_light_temp_srv_set(&srv->temp_srv, NULL, &temp, NULL);
-
-		if (!atomic_test_bit(&srv->lightness_srv.flags,
-				     LIGHTNESS_SRV_FLAG_IS_ON)) {
-			break;
-		}
-
-		light.lvl = srv->lightness_srv.last;
-		lightness_srv_change_lvl(&srv->lightness_srv, NULL, &light,
-					 NULL);
 		break;
 
 	default:
 		return -EINVAL;
 	}
 
+	if (IS_ENABLED(CONFIG_BT_MESH_MODEL_EXTENSIONS)) {
```

Comment:
Is this conditional necessary? Light CTL server cannot be used unless model extensions are enabled anyway. All such conditionals could be replaced by a single assertion in respective server headers.

Reply:
The extension API is now forced if settings are enabled. If settings are disabled, power-up behavior has no effect.

Reply:
The extension should always be forced for extended models irrespective of settings, otherwise, it alters the behavior of shared subs. So, considering that this common boilerplate if could be moved to a single assertion in the header.